rbac/init.go

```diff
@@ -11,56 +12,152 @@ import (
+			} else {
+				definitions = append(definitions, []string{acl.Principal, object, action, "allow"})
+			}
+		}
+	}
+	return definitions
+}
+
+type Policy struct {
+	Principal string   `yaml:"principal" json:"principal"`
+	ACLs      []ACL    `yaml:"acl" json:"acl"`
+	Inherit   []string `yaml:"inherit" json:"inherit"`
+}
+
+func (p Policy) GetPolicyDefintions() [][]string {
+	var definitions [][]string
+	for _, acl := range p.ACLs {
+		definitions = append(definitions, acl.GetPolicyDefinition()...)
+	}
+	return definitions
+}
+
 const (
 	modelDefinition = `
     [request_definition]
     r = sub, obj, act
 
     [policy_definition]
-    p = sub, obj, act
+    p = sub, obj, act, eft
 
     [role_definition]
     g = _, _
 
     [policy_effect]
-    e = some(where (p.eft == allow))
+		e = some(where (p.eft == allow)) && !some(where (p.eft == deny))
 
     [matchers]
-    m = g(r.sub, p.sub) && r.obj == p.obj && r.act == p.act`
+    m = g(r.sub, p.sub) && ( p.obj == '*' || r.obj == p.obj) && (p.act == '*' || r.act == p.act)`
 
 	// Roles
 	RoleAdmin     = "admin"
+	RoleEveryone  = "everyone"
 	RoleEditor    = "editor"
 	RoleViewer    = "viewer"
 	RoleCommander = "commander"
 	RoleResponder = "responder"
 	RoleAgent     = "agent"
 
 	// Actions
-	ActionRead   = "read"
-	ActionWrite  = "write"
-	ActionUpdate = "update"
-	ActionCreate = "create"
+	ActionRead    = "read"
+	ActionWrite   = "write"
+	ActionCreate  = "create"
+	ActionDelete  = "delete"
+	ActionRun     = "run"
+	ActionApprove = "approve"
+	ActionAll     = "*"
+	ActionCRUD    = "create,read,update,delete"
 
 	// Objects
-	ObjectRBAC        = "rbac"
-	ObjectAuth        = "auth"
-	ObjectAgentPush   = "agent-push"
-	ObjectAgentCreate = "agent-create"
-	ObjectDatabase    = "database"
-
-	ObjectDatabaseResponder      = "database.responder"
-	ObjectDatabaseIncident       = "database.incident"
-	ObjectDatabaseEvidence       = "database.evidences"
-	ObjectDatabaseComment        = "database.comments"
-	ObjectDatabaseCanary         = "database.canaries"
-	ObjectDatabaseSystemTemplate = "database.system_templates"
-	ObjectDatabaseConfigScraper  = "database.config_scrapers"
-	ObjectDatabaseIdentity       = "database.identities"
-	ObjectDatabaseConnection     = "database.connections"
-	ObjectDatabaseKratosTable    = "database.kratos"
+	ObjectLogs             = "logs"
+	ObjectAgent            = "agent"
+	ObjectAgentPush        = "agent-push"
+	ObjectArtifact         = "artifact"
+	ObjectAuth             = "auth"
+	ObjectCanary           = "canaries"
+	ObjectCatalog          = "catalog"
+	ObjectConnection       = "connection"
+	ObjectDatabase         = "database"
+	ObjectDatabaseIdentity = "database.identities"
+	ObjectAuthConfidential = "database.kratos"
+	ObjectDatabasePublic   = "database.public"
+	ObjectDatabaseSettings = "database.config_scrapers"
+	ObjectDatabaseSystem   = "database.system"
+	ObjectIncident         = "incident"
+	ObjectMonitor          = "database.monitor"
```

Comment:
The `database.` prefix is to map directly to tables, we don't have `monitor` and `system` tables